### ISSUE TITLE
Finish renaming Piwik to Matomo everywhere

### ISSUE THREAD

**Issue by mattab** [matomo-org/matomo#12420]
In January 2018 [we announced that Piwik is now Matomo](https://matomo.org/blog/2018/01/piwik-is-now-matomo/) :tada: 

As part of the renaming efforts we're updating all our websites, tools, core app, plugins, ie. everything! 

Renaming checklist left to do:

* [ ] HTTP API methods eg. `ExampleAPI.getPiwikVersion`
* [ ] plugin `CustomPiwikJs` to be renamed eg. `CustomJsTracker`
* [ ] HTTP Tracking API endpoints: `piwik.js` and `piwik.php` 
* [ ] JavaScript tracking sdk with objects like `Piwik` and `window.Piwik` and methods like `getPiwikUrl`
* [ ] Plugin developers, core platform uses namespace `Piwik` + some objects called "Piwik", an event `Piwik.getJavascriptCode`
* [ ] cookie names eg `piwik_auth`
* [ ] URL parameter like `piwik_campaign` to also support `matomo_campaign`
* [ ] DB prefix like `piwik_`
* [ ] Marketplace URL parameters like "piwik_version"
* [ ] plugin.json parameters like require "piwik"
* [ ] ... and more

In the case of Matomo core platform we want to keep things backward compatible for users including HTTP APIs users or users of the JavaScript SDKs. 

Assigning to Matomo 4.0.0 but hopefully we can tackle a few of the list, and make progress on the others, during Matomo 3.x :rocket:

**Comment by linusg** [comment]
Hey there,
not sure if there's a separate repo for the website (matomo.org, didn't found one), but I think it will fit in here.

On the main page, there's this animation:

![image](https://user-images.githubusercontent.com/19366641/42729324-d483a52a-87d4-11e8-9b66-0149cc4a8234.png)

On the truck it say's Piwik, so that should be changed at one point in the future. :-)

Cheers!

**Comment by tsteur** [comment]
As part of this issue it be important to look at which ones are easy and quick to do. The rest we'll do as part of Matomo 5.

**Comment by tsteur** [comment]
for tracker created https://github.com/matomo-org/tracker-proxy/pull/61

**Comment by tsteur** [comment]
I reckon we can move the rest to Matomo 5? The only question would be wether we want to release another Matomo 2 and Matomo 3 update with the legacy auto loader: https://github.com/matomo-org/matomo/blob/4.x-dev/LegacyAutoloader.php#L17

This way should we ever change the namespace then it would cause less upgrade issues. 

@sgiehl @mattab I reckon it could make sense to add the autoloader and release new version also for Matomo 2.* users

**Comment by sgiehl** [comment]
@tsteur Yes. Guess it makes sense to release new versions for 2.x and 3.x with the autoloader, so we have smoother updates later...

**Comment by tsteur** [comment]
@sgiehl created this for 3.x-dev: https://github.com/matomo-org/matomo/compare/3.x-dev...legacyautoloader?expand=1 would create the same for 2.x-dev... do you reckon this would help?

Generally thinking the autloader needs to work both ways `Piwik -> Matomo` and `Matomo -> Piwik`

* Core updater runs with 3.x, all files from Matomo are being updated
* Now it tries to request `\Piwik\Http` class in https://github.com/matomo-org/matomo/blob/3.13.6/plugins/CoreUpdater/Updater.php#L131
* Because the file `core/Http` was replaced this file will contain `\Matomo\Http`... and no longer `\Piwik\Http`... 
* The error would not be solved by this legacy autoloader unless we add alias as needed both ways (which I now did in the updated code base)

On yet another note noticed there is also `interface_exists`. Hoping we won't run into any issues around interfaces but luckily we don't use them much :)

**Comment by sgiehl** [comment]
Aren't we running most of the update process on the new code base? 🤔 
But guess it makes sense to have the autoloader in both ways, to be sure there can't be any renamed class missing...

**Comment by tsteur** [comment]
FYI we basically need to release a Matomo 2 release and then we can move this issue into Matomo 5 milestone.

**Comment by rklrkl** [comment]
I just updated to Matomo 5.0.1 and post-update "Matomo Changelog" panel says "See the latest changes in Piwik.". Doing a quick scan of the Web tree, I found the HTML code in <webtree>/tmp/cache/tracker/matomocache_RSS_d88870a6de03bc454328f6b17452cb70.php so I'm guessing this is supplied by matomo.org and then cached locally. I think this might be the first time I've seen an incorrect visible textual reference to Piwik after logging in to our local Matomo instance.
